plugins/experimental/money_trace/money_trace.cc

```diff
@@ -120,17 +120,13 @@ mt_check_request_header(TSHttpTxn txnp)
       if (!hdr_value || length <= 0) {
         LOG_DEBUG("ignoring, corrupt money trace header.");
       } else {
-        txn_data                                   = allocTransactionData();
-        txn_data->client_request_mt_header         = TSstrndup(hdr_value, length);
-        txn_data->client_request_mt_header[length] = '\0'; // workaround for bug in core.
-        LOG_DEBUG("found money trace header: %s, length: %d", txn_data->client_request_mt_header, length);
         if (nullptr == (contp = TSContCreate(transaction_handler, nullptr))) {
           LOG_ERROR("failed to create the transaction handler continuation");
-          if (nullptr != txn_data) {
-            TSfree(txn_data->client_request_mt_header);
-            TSfree(txn_data);
-          }
         } else {
+          txn_data                                   = allocTransactionData();
+          txn_data->client_request_mt_header         = TSstrndup(hdr_value, length);
```

Comment:
Hey @maskit! Thanks for reviewing this.

I don't think so ... txn_data is already deferenced in allocTransactionData(), so I assume a null check here would just cause another Coverity defect -- dereference before null check. The current implementation doesn't null check txn_data and this patch doesn't change that.